integrations/QuoraPixel/browser.js

```diff
@@ -0,0 +1,67 @@
+/* eslint-disable */
+import logger from "../../utils/logUtil";
+import { NAME } from "./constants";
+import { getHashFromArrayWithDuplicate } from "../utils/commonUtils";
+
+class QuoraPixel {
+  constructor(config) {
+    this.name = NAME;
+    this.pixelId = config.pixelId;
+    this.eventsToQPEvents = config.eventsToQPEvents;
+  }
+
+  loadScript() {
+    !(function (q, e, v, n, t, s) {
+      if (q.qp) return;
+      n = q.qp = function () {
+        n.qp ? n.qp.apply(n, arguments) : n.queue.push(arguments);
+      };
+      n.queue = [];
+      t = document.createElement(e);
+      t.async = !0;
+      t.src = v;
+      s = document.getElementsByTagName(e)[0];
+      s.parentNode.insertBefore(t, s);
+    })(window, "script", "https://a.quora.com/qevents.js");
+    window.qp("init", this.pixelId);
+    window.qp("track", "ViewContent");
+  }
+
+  init() {
+    logger.debug("===In init Quora Pixel===");
+    this.loadScript();
+  }
+
+  isLoaded() {
+    logger.debug("===In isLoaded Quora Pixel===");
+    return !!(window.qp && window.qp.push !== Array.prototype.push);
+  }
+
+  isReady() {
+    logger.debug("===In isReady Quora Pixel===");
+    return !!(window.qp && window.qp.push !== Array.prototype.push);
+  }
+
+  track(rudderElement) {
+    logger.debug("===In Quora Pixel track===");
+    const { event } = rudderElement.message;
+    const eventsMapping = getHashFromArrayWithDuplicate(this.eventsToQPEvents);
+    const trimmedEvent = event.toLowerCase().trim();
+    const events = eventsMapping[trimmedEvent];
+
+    if (!events || events.length === 0) {
+      window.qp("track", "Generic");
+      return;
+    }
+
+    events.forEach((qpEvent) => {
+      if (!qpEvent) {
```

Comment:
when this condition will occur?

Reply:
If in webapp, dropdown option is not selected then `getHashFromArrayWithDuplicate` will return.
["Purchase", ""]
<img width="497" alt="Screenshot 2022-09-12 at 6 35 16 PM" src="https://user-images.githubusercontent.com/60897972/189661335-6d5e91ac-b03c-4dad-8c54-b8f2718f1105.png">

Reply:
got it. But in that case you need to check if this an empty string or not

Reply:
This will work for empty string also.
<img width="1725" alt="Screenshot 2022-09-12 at 6 44 54 PM" src="https://user-images.githubusercontent.com/60897972/189664066-068343f6-f6b3-40b6-905a-d0b98f806593.png">

Reply:
cool then. We can keep it.